notifications_utils/field.py

```diff
@@ -16,17 +18,45 @@ def __init__(self, body):
         # body shouldn’t include leading/trailing brackets, like (( and ))
         self.body = body.lstrip("(").rstrip(")")
 
+    class Types(StrEnum):
+        BASE = auto()
+        UNSAFE = auto()
+        CONDITIONAL = auto()
+
+    extended_type_pattern = OrderedDict(
+        {
+            Types.UNSAFE: "::unsafe",
```

Comment:
This will match:
- `((example::unsafefoobar))` (should probably be treated as a normal placeholder)
- `((example??foo::unsafe))` (should probably be treated as a conditional placeholder)

Reply:
type attribute now matches on a regex, added additional tests to check against this.